src/raft/uv_list.c

```diff
@@ -100,6 +100,7 @@ int UvList(struct uv *uv,
 
 	if (rv != 0 && *segments != NULL) {
 		raft_free(*segments);
+		*segments = NULL;
 	}
```

Comment:
The actual change here looks good, but upon reading the code it seems like we should also free `*snapshots` if `rv != 0`. Would you mind adding that to your PR?

Reply:
Sure, added.

Reply:
Thanks!